erizo_controller/erizoJS/models/Connection.js

```diff
@@ -52,27 +67,91 @@ class Connection {
+    }
     return mediaStream;
   }
 
-  addMediaStream(id) {
+  init() {
+    if (this.initialized) {
+      return;
+    }
+    this.initialized = true;
+    log.debug(`message: Init Connection, connectionId: ${this.id} `+
+              `${logger.objectToLog(this.options)}`);
+
+    this.wrtc.init((newStatus, mess) => {
+      log.info('message: WebRtcConnection status update, ' +
+               'id: ' + this.id + ', status: ' + newStatus +
+                ', ' + logger.objectToLog(this.metadata));
+      switch(newStatus) {
+        case CONN_INITIAL:
+          this.emit('status_event', {type: 'started'}, newStatus);
+          break;
+
+        case CONN_SDP:
+        case CONN_GATHERED:
+          this.wrtc.localDescription = new SessionDescription(this.wrtc.getLocalDescription());
+          const sdp = this.wrtc.localDescription.getSdp();
+          mess = sdp.toString();
+          mess = mess.replace(this.privateRegexp, this.publicIP);
+
+          const info = {type: this.options.createOffer ? 'offer' : 'answer', sdp: mess};
+          this.emit('status_event', info, newStatus);
+          break;
+
+        case CONN_CANDIDATE:
+          mess = mess.replace(this.privateRegexp, this.publicIP);
+          this.emit('status_event', {type: 'candidate', candidate: mess}, newStatus);
+          break;
+
+        case CONN_FAILED:
+          log.warn('message: failed the ICE process, ' + 'code: ' + WARN_BAD_CONNECTION +
+                   ', id: ' + this.id);
+          this.emit('status_event', {type: 'failed', sdp: mess}, newStatus);
+          break;
+
+        case CONN_READY:
+          log.debug('message: connection ready, ' + 'id: ' + this.id +
+                    ', ' + 'status: ' + newStatus);
+          this.emit('status_event', {type: 'ready'}, newStatus);
+          break;
+      }
+    });
+    if (this.options.createOffer) {
+      log.debug('message: create offer requested, id:', this.id);
+      const audioEnabled = this.options.createOffer.audio;
+      const videoEnabled = this.options.createOffer.video;
+      const bundle = this.options.createOffer.bundle;
+      this.wrtc.createOffer(videoEnabled, audioEnabled, bundle);
+    }
+    this.emit('status_event', {type: 'initializing'});
+  }
+
+  addMediaStream(id, mediaStreamOptions) {
     log.info(`message: addMediaStream, connectionId: ${this.id}, mediaStreamId: ${id}`);
     if (this.mediaStreams.get(id) === undefined) {
-      const mediaStream = this._createMediaStream(id);
+      const mediaStream = this._createMediaStream(id, mediaStreamOptions);
       this.wrtc.addMediaStream(mediaStream);
       this.mediaStreams.set(id, mediaStream);
     }
   }
+
   removeMediaStream(id) {
-    if(this.mediaStreams.get(id) !== undefined) {
+    if (this.mediaStreams.get(id) !== undefined) {
       this.wrtc.removeMediaStream(id);
       this.mediaStreams.get(id).close();
       this.mediaStreams.delete(id);
```

Comment:
Maybe use `getNumMediaStreams` instead? Otherwise we don't really need it at the moment

Reply:
I mean in the log message [below](https://github.com/lynckia/licode/pull/1164/files/b2b244865ad336fbcefe14ec01e3acd4eb57e05f#diff-a257b3a662e2e651ddb5db4d1565d303L79)

Reply:
good catch!